src/main/java/io/cryostat/agent/ConfigModule.java

```diff
@@ -308,13 +306,14 @@ public static String provideCryostatAgentWebclientTlsTruststorePassCharset(Confi
     public static Optional<ByteBuffer> provideCryostatAgentWebclientTlsTruststorePass(
             Config config,
             @Named(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS_FILE)
-                    Optional<ByteBuffer> truststorePass) {
+                    Optional<ByteBuffer> truststorePass,
+            @Named(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS_CHARSET) String passCharset) {
         Optional<String> opt =
                 config.getOptionalValue(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS, String.class);
```

Comment:
I don't suppose swapping `String.class` for `byte[].class` is supported by microprofile-config, is it?

Reply:
I can make a converter so in ConfigModule I can extract `byte[]` or even `ByteBuffer` type, but it would still use `config.getOptionalValue(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS, String.class` in the converter itself.

Reply:
I figured. Not worth doing in that case - I'm just trying to see if there's any feasible way for us to get around having to ever read this value as a String.

Anyway, supplying the password as a mounted volume/projected file is already the better security practice, so if we're handling that case properly with a cleared byte array, then I think the bases are covered for the real security aspects.